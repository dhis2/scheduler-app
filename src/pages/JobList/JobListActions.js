import React from 'react'
import { PropTypes } from '@dhis2/prop-types'
import { Menu, DropdownButton } from '@dhis2/ui'
import i18n from '@dhis2/d2-i18n'
import EditJobMenuItem from './EditJobMenuItem'
import RunJobMenuItem from './RunJobMenuItem'
import DeleteJobMenuItem from './DeleteJobMenuItem'

const JobListActions = ({ id }) => (
    <DropdownButton
        small
        component={
            <Menu>
                <EditJobMenuItem id={id} />
                <RunJobMenuItem id={id} />
                <DeleteJobMenuItem id={id} />
            </Menu>
        }
    >
        {i18n.t('Actions')}
    </DropdownButton>
)

const { string } = PropTypes

JobListActions.propTypes = {
    id: string.isRequired,
}

export default JobListActions
