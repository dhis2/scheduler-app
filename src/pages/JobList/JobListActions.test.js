import React from 'react'
import { shallow } from 'enzyme'
import JobListActions from './JobListActions'

describe('<JobListActions>', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<JobListActions id="1" />)

        expect(wrapper).toMatchSnapshot()
    })
})
