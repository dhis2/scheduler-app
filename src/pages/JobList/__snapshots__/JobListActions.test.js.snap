// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobListActions> renders correctly 1`] = `
<DropdownButton
  component={
    <Menu
      dataTest="dhis2-uicore-menu"
      maxWidth="380px"
    >
      <EditJobMenuItem
        id="1"
      />
      <Memo(ConnectFunction)
        id="1"
      />
      <Memo(ConnectFunction)
        id="1"
      />
    </Menu>
  }
  dataTest="dhis2-uicore-dropdownbutton"
  small={true}
>
  Actions
</DropdownButton>
`;
